netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -634,6 +640,9 @@ protected void userEventTriggered0(ChannelHandlerContext ctx, Object evt) throws
                 .withCause(t)
                 .asRuntimeException();
           }
+          if (handshakeCompleteRunnable.isPresent()) {
+            handshakeCompleteRunnable.get().run();
+          }
           ctx.fireExceptionCaught(t);
```

Comment:
There's one more fireExceptionCaught in this method, up on line 623/629.

Reply:
Done in https://github.com/grpc/grpc-java/pull/11600/commits/6cce1561c84905d3435b9e76195dd7b9bb25b93a